src/wq/ql/flaskutils.py

```diff
@@ -12,10 +12,7 @@ def json_array_streamer(xs: Iterable[BaseModel]) -> Iterable[str]:
     for x in xs:
         json_repr = x.json()
         yield json_repr + ',\n'
-    yield 'null\n]'
-# TODO how to get rid of the null terminator in an efficient and **simple**
-# way? I could use the same put-back approach as in itersplit but I'd rather
-# keep it simple.
+    yield '\n]'
```

Comment:
@NEC-Vishal nice try, but it won't work :-)
In fact, if the input iterable isn't empty the output JSON array will contain an extra comma before the array end symbol, e.g. `[1, 2, 3] ~~~> "[\n1,\n2,\n3,\n]"` which is **not** a valid JSON array!